### PR TITLE
fix(massive-downtime): issue fixed with GET entry

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

🏷️ MON-16773

Issue fixed when using the 'select' GET entry.

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [x] 21.10.x
- [x] 22.04.x
- [x] 22.10.x
- [x] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

See jira ticket

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/include/monitoring/external_cmd/popup/massive_downtime.php [hunk 1]
```diff
@@ -38,14 +38,16 @@
...
 
+const DOWNTIME_ON_HOST = 75;
+
 $select = array();
 if (isset($_GET['select'])) {
     foreach ($_GET['select'] as $key => $value) {
-        if ($cmd == '75') {
-            $tmp = preg_split("/\;/", $key);
+        if ((int) $cmd == DOWNTIME_ON_HOST) {
+            $tmp = preg_split("/\;/", urlencode($key));
```
Comment (abuathier) [inline centreon/www/include/monitoring/external_cmd/popup/massive_downtime.php:47]: [urlencode](https://www.php.net/manual/en/function.urlencode.php) ? not [rawurlencode](https://www.php.net/manual/en/function.rawurlencode.php) (RFC 3986) ?

File: centreon/www/include/monitoring/external_cmd/popup/massive_downtime.php [hunk 2]
```diff
@@ -38,14 +38,16 @@
...
             $select[] = $tmp[0];
         } else {
-            $select[] = $key;
+            $select[] = urlencode($key);
```
Comment (abuathier) [inline centreon/www/include/monitoring/external_cmd/popup/massive_downtime.php:50]: [urlencode](https://www.php.net/manual/en/function.urlencode.php) ? not [rawurlencode](https://www.php.net/manual/en/function.rawurlencode.php) (RFC 3986) ?

Comment (adr-mo) [reply centreon/www/include/monitoring/external_cmd/popup/massive_downtime.php:50]: To be honest I've applied the same "code" has the one made for the massive_ack. Did not want to risk any side effects

**Review by abuathier — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

File: centreon/www/include/monitoring/external_cmd/popup/popup.php [hunk 3]
```diff
@@ -42,6 +42,11 @@
 require_once _CENTREON_PATH_ . "www/class/centreonLang.class.php";
 require_once _CENTREON_PATH_ . "www/include/common/common-Func.php";
 
+const AKNOWLEDGEMENT_ON_SERVICE = 70;
+const AKNOWLEDGEMENT_ON_HOST = 72;
```
Comment (a-launois) [inline centreon/www/include/monitoring/external_cmd/popup/popup.php:46]:
```suggestion
const ACKNOWLEDGEMENT_ON_SERVICE = 70;
const ACKNOWLEDGEMENT_ON_HOST = 72;
```

**Review by a-launois — CHANGES_REQUESTED** [review]

**Review by a-launois — APPROVED** [review]

**Review by abuathier — APPROVED** [review]

**Review by wtermellil — APPROVED** [review]

**Pull request merged**